web/viewer.js

```diff
@@ -297,6 +297,12 @@ var PDFViewerApplication = {
         }
         PDFJS.externalLinkTarget = value;
       }),
+      Preferences.get('withCredentials').then(function resolved(value) {
+        if (PDFJS.withCredentials === true) {
+          return;
+        }
+        PDFJS.withCredentials = value;
```

Comment:
PDFJS.withCredentials is not part of the api.js. It will be better if this preference will stay as a part of UI (PDFViewerApplication) and be passed in the open() to the getDocument call.